lib/smart_answer/money.rb

```diff
@@ -32,6 +29,23 @@ def <=>(other)
       end
     end
 
+    def self.raw_input_to_number(raw_input)  
+      if raw_input.is_a?(Numeric)
+        raw_input.to_f.infinite? ? 
+          [nil, "Sorry, that number is too big. Please try again."] :
+          [raw_input, nil]
+      else
+        raw_input = raw_input.to_s.delete(',').gsub(/\s/, '')
+        if raw_input.to_f.infinite?
```

Comment:
As per my previous comment, you could avoid having to check `infinite?` multiple times if you just call it once when you _know_ you have a `Numeric` (probably outside this method).